File: backend/src/main/java/top/smartduck/ducktodo/model/entity/TaskAudit.java
package top.smartduck.ducktodo.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 任务审计实体，对应表 `task_audit`。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("task_audit")
public class TaskAudit implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 审计日志id（UUID）
     */
    @TableId(value = "audit_id", type = IdType.ASSIGN_UUID)
    private String auditId;

    /**
     * 关联的主任务id（UUID）
     */
    private String taskId;

    /**
     * 操作人id（UUID）
     */
    private String operatorId;

    /**
     * 操作类型（CREATE, UPDATE, DELETE, COMPLETE...）
     */
    private String actionType;

    /**
     * 操作详情
     */
    private String actionDescription;

    /**
     * 是否删除 0-未删除 1-已删除
     */
    @TableLogic
    private Integer isDelete;

    /**
     * 操作时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}

File: ai-backend/models/entity/task_audit.py
"""
任务审计实体，对应表 `task_audit`
"""
from typing import Optional
from datetime import datetime
from models.entity.base import BaseEntity


class TaskAudit(BaseEntity):
    """任务审计实体"""
    audit_id: Optional[str] = None  # 审计日志id（UUID）
    task_id: Optional[str] = None  # 关联的主任务id（UUID）
    operator_id: Optional[str] = None  # 操作人id（UUID）
    action_type: Optional[str] = None  # 操作类型（CREATE, UPDATE, DELETE, COMPLETE...）
    action_description: Optional[str] = None  # 操作详情
    
    class Config:
        from_attributes = True
